frost-core/src/tests/refresh.rs

```diff
@@ -0,0 +1,142 @@
+//! Test for Refreshing shares
+
+use std::collections::BTreeMap;
+
+use rand_core::{CryptoRng, RngCore};
+
+use crate::keys::refresh::{calculate_zero_key, refresh_share};
+use crate::{self as frost};
+use crate::{
+    keys::{PublicKeyPackage, SecretShare},
+    Ciphersuite, Error, Identifier, SigningKey,
+};
+
+use super::ciphersuite_generic::check_sign;
+
+/// We want to test that recover share matches the original share
+pub fn check_refresh_shares_with_dealer<C: Ciphersuite, R: RngCore + CryptoRng>(mut rng: R) {
+    // Compute shares
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Old Key generation
+    ////////////////////////////////////////////////////////////////////////////
+
+    let max_signers = 5;
+    let min_signers = 3;
+    let (old_shares, pub_key_package) = frost::keys::generate_with_dealer(
+        max_signers,
+        min_signers,
+        frost::keys::IdentifierList::Default,
+        &mut rng,
+    )
+    .unwrap();
+
+    // Try to refresh shares
+    // Signer 2 will be removed and Signers 1, 3, 4 & 5 will remain
+
+    ////////////////////////////////////////////////////////////////////////////
+    // New Key generation
+    ////////////////////////////////////////////////////////////////////////////
+
+    let remaining_ids = vec![
+        Identifier::try_from(1).unwrap(),
+        Identifier::try_from(3).unwrap(),
+        Identifier::try_from(4).unwrap(),
+        Identifier::try_from(5).unwrap(),
+    ];
+
+    const NEW_MAX_SIGNERS: u16 = 4;
+
+    // Trusted Dealer generates zero keys
+
+    let (zero_shares, new_pub_key_package) = calculate_zero_key(
+        pub_key_package,
+        NEW_MAX_SIGNERS,
+        min_signers,
+        &remaining_ids,
+        &mut rng,
+    )
+    .unwrap();
+
```

Comment:
At this point we can do a serialization round trip to make sure we are not serializing the identity.